aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -810,6 +810,19 @@ Tensor masked_select_cpu(const Tensor & self, const Tensor & mask) {
   return masked_select_out_cpu(result, self, mask);
 }
 
+Tensor masked_select_backward(const Tensor& grad, const Tensor& input, const Tensor& mask) {
+  // normally broadcasting is handled implicitly, but here, because we call an inplace
```

Comment:
I don't think the reason for the inplace here is optimization. We cannot really do it any other way as we need to have 0s for the unused values.

Reply:
Thanks for the catch! I was copying-pasting straight from the original code, but I agree it's not really an optimization. I'll delete the "as an optimization" part of the comment.

Reply:
Discussed offline: there actually is an optimization going on (we could have just done `zeros_like(input).masked_scatter(mask, grad)`. I clarified the comment with the example